src/main/resources/config/handler.yml

```diff
@@ -83,18 +83,25 @@ paths:
     exec:
       - admin
       - info
-  # the health endpoint for the proxy itself without security, this for both Kubernetes liveness and readiness.
+  # the health endpoint for the proxy itself without security, this for both Kubernetes liveness and readiness of sidecar. It does NOT proxy health to downstream backend.
   - path: '/health'
     method: 'get'
     exec:
       - exception
       - health
-  # the health check endpoint called from control plane with security and it will invoke backend optionally.
-  - path: '/adm/health/${server.serviceId}'
+  # the health endpoint for the proxy + downstream backend if enabled without security, this is for teams to monitor overall API health via third party tools or via Control Plane
+  - path: '/health/${server.serviceId}'
     method: 'get'
     exec:
-      - admin
```

Comment:
If the health endpoint is proxied to the backend, should we protect it with the admin chain for security? If we don't, we might have the risk of a DDoS attack to send too many requests to the backend. At least we need a whitelist to allow only certain clients to connect. What do you think?

Reply:
Interesting point. Few counter-points:

- The risk of giving health monitor services the admin credentials might outweigh the risk of DDoS;
- There's an assumption that proxy sidecar can absorb more request volume since DDoS attacks will affect non-proxied `/health` regardless and the microservice as a unit would be unavailable anyway;
- There's also higher complexity (and cost) associated to authentication mechanism for frequent health checks (teams might start disabling security of `/health` altogether to avoid the extra overhead for monitors within same network zone);
- DDoS can be mitigated by configuring default Rate Limit for `/health` based on resource allocation;